app/src/main/java/com/mapbox/maps/testapp/examples/markersandcallouts/ViewAnnotationBasicAddActivity.kt

```diff
@@ -0,0 +1,79 @@
+package com.mapbox.maps.testapp.examples.markersandcallouts
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.widget.Button
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.geojson.Point
+import com.mapbox.maps.*
+import com.mapbox.maps.plugin.gestures.*
+import com.mapbox.maps.testapp.R
+import com.mapbox.maps.testapp.databinding.ActivityViewAnnotationShowcaseBinding
+import com.mapbox.maps.testapp.databinding.ItemCalloutViewBinding
+import com.mapbox.maps.viewannotation.ViewAnnotationManager
+import com.mapbox.maps.viewannotation.viewAnnotationOptions
+
+/**
+ * Example how to add view annotations by clicking on the map.
+ */
+class ViewAnnotationBasicAddActivity : AppCompatActivity(), OnMapClickListener {
+
+  private lateinit var mapboxMap: MapboxMap
+  private lateinit var viewAnnotationManager: ViewAnnotationManager
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val binding = ActivityViewAnnotationShowcaseBinding.inflate(layoutInflater)
+    setContentView(binding.root)
+
+    viewAnnotationManager = binding.mapView.viewAnnotationManager
```

Comment:
should we deallocate when `onDestroy` happens?
also question about mapboxMap? set to null

Reply:
it's deallocated automatically when `MapView` is destroyed which happens automatically as we have lifecycle-plugin enabled.

Reply:
see https://github.com/mapbox/mapbox-maps-android/blob/8e3507cff85e22d0a670e3606ec5fd595b9f63bb/sdk/src/main/java/com/mapbox/maps/MapView.kt#L214-L216